File: example/src/main/java/com/thesurix/example/gesturerecycler/callback/MonthDiffCallback.java
package com.thesurix.example.gesturerecycler.callback;


import com.thesurix.example.gesturerecycler.model.MonthItem;

import android.support.v7.util.DiffUtil;

import java.util.List;

public class MonthDiffCallback extends DiffUtil.Callback {

    private final List<MonthItem> mOldList;
    private final List<MonthItem> mNewList;

    public MonthDiffCallback(final List<MonthItem> oldList, final List<MonthItem> newList) {
        mOldList = oldList;
        mNewList = newList;
    }

    @Override
    public int getOldListSize() {
        return mOldList == null ? 0 : mOldList.size();
    }

    @Override
    public int getNewListSize() {
        return mNewList == null ? 0 : mNewList.size();
    }

    @Override
    public boolean areItemsTheSame(final int oldItemPosition, final int newItemPosition) {
        return areItemsEqual(oldItemPosition, newItemPosition);
    }

    @Override
    public boolean areContentsTheSame(final int oldItemPosition, final int newItemPosition) {
        return areItemsEqual(oldItemPosition, newItemPosition);
    }

    private boolean areItemsEqual(final int oldItemPosition, final int newItemPosition) {
        final MonthItem oldItem = mOldList.get(oldItemPosition);
        final MonthItem newItem = mNewList.get(newItemPosition);

        return oldItem.getName().equals(newItem.getName());
    }
}

File: example/src/main/java/com/thesurix/example/gesturerecycler/callback/MonthDiffCallback.kt
package com.thesurix.example.gesturerecycler.callback


import androidx.recyclerview.widget.DiffUtil
import com.thesurix.example.gesturerecycler.model.MonthItem

class MonthDiffCallback(private val mOldList: List<MonthItem>, private val mNewList: List<MonthItem>) : DiffUtil.Callback() {

    override fun getOldListSize(): Int {
        return mOldList.size
    }

    override fun getNewListSize(): Int {
        return mNewList.size
    }

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return areItemsEqual(oldItemPosition, newItemPosition)
    }

    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return areItemsEqual(oldItemPosition, newItemPosition)
    }

    private fun areItemsEqual(oldItemPosition: Int, newItemPosition: Int): Boolean {
        val oldItem = mOldList[oldItemPosition]
        val newItem = mNewList[newItemPosition]

        return oldItem.name == newItem.name
    }
}
